cypress/e2e/tests/administration/custom-migration-targets/crud.test.ts

```diff
@@ -42,248 +42,95 @@ import { cancelButton } from "../../../views/common.view";
-            this.customMigrationTargets = customMigrationTargets;
-        });
-
-        cy.fixture("application").then(function (appData) {
-            this.appData = appData;
-        });
-
-        cy.fixture("analysis").then(function (analysisData) {
-            this.analysisData = analysisData;
-        });
-
-        cy.intercept("POST", "/hub/targets/*").as("postRule");
-        cy.intercept("GET", "/hub/targets*").as("getRule");
-        cy.intercept("PUT", "/hub/targets*/*").as("putRule");
-        cy.intercept("DELETE", "/hub/targets*/*").as("deleteRule");
-
-        CustomMigrationTarget.open(true);
-    });
-
     Object.values(Languages).forEach((language) => {
-        it(`${language} | Custom Migration Targets CRUD with rules uploaded manually`, function () {
-            const targetData = this.customMigrationTargets["manual_rules"];
-            const target = new CustomMigrationTarget(
-                data.getRandomWord(8),
-                data.getDescription(),
-                targetData.image,
-                getRulesData(targetData),
-                language
-            );
-            target.create();
-            cy.wait(2 * SEC);
-            cy.contains(CustomMigrationTargetView.takeMeThereNotification).click();
-            closeSuccessAlert();
-            cy.get(CustomMigrationTargetView.card, { timeout: 12 * SEC }).should(
-                "contain",
-                target.name
-            );
-
-            const newName = data.getRandomWord(8);
-            const newRules = {
-                ...target.ruleTypeData,
-                rulesetPaths: ["xml/javax-package-custom.windup.xml"],
-            };
-
-            target.edit({
-                name: newName,
-                ruleTypeData: newRules,
-            });
-            cy.get(CustomMigrationTargetView.card, { timeout: 12 * SEC }).should(
-                "contain",
-                newName
-            );
-            target.name = newName;
-            target.ruleTypeData = newRules;
-
-            target.delete();
-            cy.wait("@deleteRule");
-            cy.wait(3 * SEC);
-            cy.get(CustomMigrationTargetView.cardContainer).then((container) => {
-                if (container.children().length > 1) {
-                    cy.get(CustomMigrationTargetView.card, { timeout: 12 * SEC }).should(
-                        "not.contain",
-                        target.name
-                    );
+        describe(`${language} language`, function () {
+            // Automates Polarion TC 300 & 305
+            beforeEach("Fixtures and Interceptors", function () {
+                appFixture = "application";
+                analysisFixture = "analysis";
+                manualRulefile = "manual_rules";
+
+                if (language === Languages.Go) {
+                    appFixture = "go-application";
+                    analysisFixture = "go-analysis";
+                    manualRulefile = "manual_golang_target";
```

Comment:
If you don't want to add an `if` for each new language, you could just set the language name as the prefix of the file, like you did for `appFixture` and `analysisFixture`.

So you could do something like:


```typescript
if (language !== Languages.Java) {
    languageLower = language.toLowerCase();
    appFixture = `${languageLower}-application`;
    analysisFixture = `${languageLower}-analysis`;
    manualRulefile = `${languageLower}-manual-target`;
}
```

If all the elements are correctly named, this should work for all new languages

Reply:
Nice!! Thank!